diff --git a/plugins/sparql_views_plugin_pager_full.inc b/plugins/sparql_views_plugin_pager_full.inc
index 613171e..88db379 100644
--- a/plugins/sparql_views_plugin_pager_full.inc
+++ b/plugins/sparql_views_plugin_pager_full.inc
@@ -8,10 +8,15 @@
 
 class sparql_views_plugin_pager_full extends views_plugin_pager_full {
   /**
-   * Execute the count query, which will be done just prior to the query
-   * itself being executed.
+   * Overrides views_plugin_pager_full::execute_count_query().
+   *
+   * This is done so that SPARQL Views can swap in its own COUNT handling. If
+   * this is not a SPARQL View, then call the parent instead.
    */
   function execute_count_query(&$count_query) {
+    if (get_class($this->view->query) !== 'sparql_views_plugin_query_sparql') {
+      return parent::execute_count_query($count_query);
+    }
     $sparql_result = sparql_request($count_query, $this->view->query->endpoint);
     if (empty($sparql_result) || empty($sparql_result['result']['rows'])) {
       vsm(t('No result returned. Please check your query and the endpoint status.'));
diff --git a/plugins/sparql_views_plugin_pager_mini.inc b/plugins/sparql_views_plugin_pager_mini.inc
index 2ab3ff1..e58e287 100644
--- a/plugins/sparql_views_plugin_pager_mini.inc
+++ b/plugins/sparql_views_plugin_pager_mini.inc
@@ -8,10 +8,15 @@
 
 class sparql_views_plugin_pager_mini extends views_plugin_pager_mini {
   /**
-   * Execute the count query, which will be done just prior to the query
-   * itself being executed.
+   * Overrides views_plugin_pager_full::execute_count_query().
+   *
+   * This is done so that SPARQL Views can swap in its own COUNT handling. If
+   * this is not a SPARQL View, then call the parent instead.
    */
   function execute_count_query(&$count_query) {
+    if (get_class($this->view->query) !== 'sparql_views_plugin_query_sparql') {
+      return parent::execute_count_query($count_query);
+    }
     $sparql_result = sparql_request($count_query, $this->view->query->endpoint);
     if (empty($sparql_result) || empty($sparql_result['result']['rows'])) {
       vsm(t('No result returned. Please check your query and the endpoint status.'));
